addons/sale/tests/test_payment_flow.py

```diff
@@ -288,3 +288,48 @@ def test_invoice_is_final(self):
             tx._reconcile_after_done()
 
         self.assertTrue(_create_invoices_mock.call_args.kwargs['final'])
+
+    def test_partial_payment_confirm_order(self):
+        """
+        Test that a sale order can be confirmed through partial payments and that
+        correct mails are sent each time.
+        """
+
+        self.amount = self.sale_order.amount_total / 2
+
+        with patch(
+            'odoo.addons.sale.models.sale_order.SaleOrder._send_payment_succeeded_for_order_mail',
+        ) as payment_succeeded_mail_mock, patch(
+            'odoo.addons.sale.models.sale_order.SaleOrder._send_order_confirmation_mail',
+        ) as confirmation_mail_mock:
```

Comment:
Would be clearer to check the calls to `_send_order_notification_mail` instead, and verify the input is the expected mail template.  That way, the call_count would be 1 & 2, which would be way more understandable when reading the test.